pkg/controller/v1alpha1/inferencegraph/knative_reconciler.go

```diff
@@ -152,6 +153,12 @@ func createKnativeService(componentMeta metav1.ObjectMeta, graph *v1alpha1api.In
 										"--graph-json",
 										string(bytes),
 									},
+									Env: []v1.EnvVar{
+										{
+											Name:  "PROPAGATE_HEADERS",
+											Value: strings.Join(config.Headers["propagate"], ","),
```

Comment:
What happens when `propagate` does not exist ?

Reply:
Value of "PROPAGATE_HEADERS" is empty string in that case.

Reply:
I think it is a bit confusing here, if  `propagate` does not exist then we should not inject this env variable

Reply:
Addressed. Adding this env variable only when it is added in router's config.